src/sw.js

```diff
@@ -208,46 +226,3 @@ setCatchHandler(({ event }) => {
       return Response.error();
   }
 });
-
-// HTML caching strategy
-const htmlStrategy = new StaleWhileRevalidate({
-  cacheName: 'pages-cache',
-  plugins: [
-    navigationNormalizationPlugin,
-    new CacheableResponsePlugin({
-      headers: {
-        'Content-Type': 'text/html; charset=UTF-8',
-      },
-      statuses: [200, 301, 404, 0],
-    }),
-  ],
-});
-
-/**
- * HTML Handler
- * @param {object} param0 - Workbox Handler
- * @return {object} - Either a redirection response or a Workbox response
- */
-async function htmlHandler({ event }) {
-  const lang = await preferences.get('lang');
-  const { request } = event;
-
-  const urlLang = request.url.replace(self.location.origin, '').split('/');
-  const currentLang = urlLang[1];
-  const isALanguage = languages.includes(currentLang);
-  const isRightLanguage = currentLang !== lang;
-  const shouldRefresh = new URL(request.url).searchParams.get('locale_fallback') !== 'true';
-
-  if (isALanguage && isRightLanguage && shouldRefresh) {
-    console.log('⇒ Redirecting');
-    urlLang[1] = lang;
-    const redirectURL = `${self.location.origin}${urlLang.join('/')}`;
-    return Response.redirect(redirectURL, 302);
-  }
-
-  try {
-    return await htmlStrategy.makeRequest({ request });
-  } catch (error) {
-    return caches.match('/404/');
-  }
```

Comment:
It looks like my 404 catch didn't make it to your update code. Was this missed or am I missing where this got moved to?

Reply:
The same logic that's already inside of `setCatchHandler()` should be sufficient.

(Though I don't know when you're actually caching the `/404/` page, so that part needs to be tweaked, probably.)